server/ai_mediaserver.go

```diff
@@ -452,6 +455,48 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
 	})
 }
 
+func (ls *LivepeerServer) UpdateLiveVideo() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		if r.Method != http.MethodPost {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		// Get stream from path param
+		stream := r.PathValue("stream")
+		if stream == "" {
+			http.Error(w, "Missing stream name", http.StatusBadRequest)
+			return
+		}
+		ls.LivepeerNode.LiveMu.RLock()
+		defer ls.LivepeerNode.LiveMu.RUnlock()
+		p, ok := ls.LivepeerNode.LivePipelines[stream]
+		if !ok {
+			// Stream not found
+			http.Error(w, "Stream not found", http.StatusNotFound)
+			return
+		}
+		defer r.Body.Close()
+		params, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		clog.V(6).Infof(ctx, "Sending Live Video Update Control API stream=%s, params=%s", stream, string(params))
+		if err := p.ControlPub.Write(strings.NewReader(string(params))); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	})
+}
+
+func (ls *LivepeerServer) cleanupLive(stream string) {
+	ls.LivepeerNode.LiveMu.Lock()
+	defer ls.LivepeerNode.LiveMu.Unlock()
+	delete(ls.LivepeerNode.LivePipelines, stream)
```

Comment:
Do we need to close the trickle publisher or sth like that?

Reply:
I think that no, IMO it should be deleted on the Orchestrator's side. We should not depend on closing the trickle channel from Gateway

Reply:
I agree that we might not want to depend on closing the control channel from the gateway. But we also might want to give the orchestrator a heads-up that the session is complete, so it can release resources attached to the job. One easy way to do this would be to close the _publish_ channel which we do here

https://github.com/livepeer/go-livepeer/blob/6a8fa52c87c4a1fab11d6dce1db3f73b7d8410f2/server/ai_live_video.go#L23-L28

However, the general tear-down process between GOWR is something that still needs some work. Especially transitions when switching Os ... but we can figure that out after the launch when we are ready to tackle selection and related work.

Reply:
Ok, added closing the channel. But yeah, I share the same feelings as you that we should work on the tear-down process. Plus we need some retries process for resilience.